debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerStreamingChangeEventSource.java

```diff
@@ -269,33 +289,43 @@ public void execute(ChangeEventSourceContext context) throws InterruptedExceptio
                                                     clock));
                             tableWithSmallestLsn.next();
                         }
-                    });
+                    }, partition.getDatabaseName());
                     lastProcessedPosition = TxLogPosition.valueOf(maxLsnResult.getMaxLsn());
                     // Terminate the transaction otherwise CDC could not be disabled for tables
                     dataConnection.rollback();
                 }
                 catch (SQLException e) {
-                    tablesSlot.set(processErrorFromChangeTableQuery(e, tablesSlot.get()));
+                    tablesSlot.set(processErrorFromChangeTableQuery(e, tablesSlot.get(), partition));
                 }
             }
+
+            offsetContext.saveStreamingExecutionContext(schemaChangeCheckpoints, tablesSlot, lastProcessedPositionOnStart, lastProcessedEventSerialNoOnStart,
+                    lastProcessedPosition, changesStoppedBeingMonotonic, shouldIncreaseFromLsn,
+                    SqlServerStreamingExecutionState.StreamingResultStatus.CHANGES_IN_DATABASE);
         }
         catch (Exception e) {
             errorHandler.setProducerThrowable(e);
         }
+
+        return new StreamingResult(offsetContext);
     }
 
-    private void migrateTable(final Queue<SqlServerChangeTable> schemaChangeCheckpoints)
+    private void migrateTable(final Queue<SqlServerChangeTable> schemaChangeCheckpoints, SqlServerOffsetContext offsetContext, SqlServerTaskPartition partition)
             throws InterruptedException, SQLException {
         final SqlServerChangeTable newTable = schemaChangeCheckpoints.poll();
         LOGGER.info("Migrating schema to {}", newTable);
-        Table tableSchema = metadataConnection.getTableSchemaFromTable(newTable);
+        Table tableSchema = metadataConnection.getTableSchemaFromTable(newTable, partition.getDatabaseName());
         dispatcher.dispatchSchemaChangeEvent(newTable.getSourceTableId(),
                 new SqlServerSchemaChangeEventEmitter(offsetContext, newTable, tableSchema, SchemaChangeEventType.ALTER));
         newTable.setSourceTable(tableSchema);
     }
 
-    private SqlServerChangeTable[] processErrorFromChangeTableQuery(SQLException exception, SqlServerChangeTable[] currentChangeTables) throws Exception {
-        final Matcher m = MISSING_CDC_FUNCTION_CHANGES_ERROR.matcher(exception.getMessage());
+    private SqlServerChangeTable[] processErrorFromChangeTableQuery(SQLException exception, SqlServerChangeTable[] currentChangeTables,
+                                                                    SqlServerTaskPartition partition)
+            throws Exception {
+        final String pattern = "Invalid object name '[#db].cdc.fn_cdc_get_all_changes_(.*)'\\."
```

Comment:
Let's try and centralize queries in one place. I think there's something elsewhere already.